pulsar-broker/src/main/java/org/apache/pulsar/broker/stats/prometheus/PrometheusMetricStreams.java

```diff
@@ -41,7 +41,11 @@ void writeSample(String metricName, Number value, String... labelsAndValuesArray
         SimpleTextOutputStream stream = initGaugeType(metricName);
         stream.write(metricName).write('{');
         for (int i = 0; i < labelsAndValuesArray.length; i += 2) {
-            stream.write(labelsAndValuesArray[i]).write("=\"").write(labelsAndValuesArray[i + 1]).write('\"');
+            String labelValue = labelsAndValuesArray[i + 1];
+            if (labelValue != null) {
+                labelValue = labelValue.replace("\"", "\\\"");
```

Comment:
Did you test this with prometheus? Their documentation doesn't indicate that escaping a forbidden character is available.

https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels

> Label names may contain ASCII letters, numbers, as well as underscores. They must match the regex [a-zA-Z_][a-zA-Z0-9_]*. Label names beginning with __ are reserved for internal use.

Reply:
That restriction is for the label name.  There is no such restriction for label value.
> Label values may contain any Unicode characters.

But +1 to testing regardless.

Reply:
Of course, thanks for the explanation @pgier and @nicoloboschi. I was mixing up my terms.